src/OrbitGl/TrackManagerTest.cpp

```diff
@@ -0,0 +1,135 @@
+  callstack_info.set_type(orbit_client_protos::CallstackInfo_CallstackType_kComplete);
+  capture_data->AddUniqueCallstack(kCallstackId, std::move(callstack_info));
+
+  // CallstackEvent
+  orbit_client_protos::CallstackEvent callstack_event;
+  callstack_event.set_callstack_id(kCallstackId);
+  callstack_event.set_thread_id(kThreadId);
+  capture_data->AddCallstackEvent(std::move(callstack_event));
+
+  capture_data->AddOrAssignThreadName(kThreadId, kThreadName);
+
+  return capture_data;
+}
+
+}  // namespace
+
+namespace orbit_gl {
+
+class UnitTestTrack : public Track {
+ public:
+  explicit UnitTestTrack(TimeGraphLayout* layout, Track::Type type, const std::string& name)
+      : Track(nullptr, nullptr, nullptr, layout, nullptr, 0), name_(name), type_(type){};
+
+  [[nodiscard]] Type GetType() const override { return type_; }
+  [[nodiscard]] float GetHeight() const override { return 100.f; }
+  [[nodiscard]] bool IsEmpty() const override { return false; }
+
+  [[nodiscard]] std::vector<std::shared_ptr<TimerChain>> GetAllChains() const override {
+    return {};
+  }
+  [[nodiscard]] std::vector<std::shared_ptr<TimerChain>> GetAllSerializableChains() const override {
+    return {};
+  }
+
+ private:
+  std::string name_;
+  Type type_;
+};  // namespace orbit_gl
+
+class TrackManagerTest : public ::testing::Test {
+ protected:
+  void SetUp() override {
+    capture_data_ = GenerateTestCaptureData();
+    track_manager_ =
+        std::make_unique<TrackManager>(nullptr, nullptr, &layout_, nullptr, capture_data_.get());
+  }
+
+  void CreateAndFillTracks() {
+    auto* scheduler_track = track_manager_->GetOrCreateSchedulerTrack();
+    auto* thread_track = track_manager_->GetOrCreateThreadTrack(kThreadId);
+
+    TimerInfo timer;
+    timer.set_start(0);
+    timer.set_end(100);
+    timer.set_thread_id(kThreadId);
+    timer.set_processor(0);
+    timer.set_depth(0);
+    timer.set_type(TimerInfo::kCoreActivity);
+
+    scheduler_track->OnTimer(timer);
+    timer.set_type(TimerInfo::kCoreActivity);
+    thread_track->OnTimer(timer);
+  }
+
+  TimeGraphLayout layout_;
+  std::unique_ptr<orbit_client_model::CaptureData> capture_data_;
+  std::unique_ptr<TrackManager> track_manager_;
+};
+
+TEST_F(TrackManagerTest, GetOrCreateCreatesTracks) {
+  EXPECT_EQ(0ull, track_manager_->GetAllTracks().size());
+
+  track_manager_->GetOrCreateSchedulerTrack();
+  EXPECT_EQ(1ull, track_manager_->GetAllTracks().size());
+  track_manager_->GetOrCreateThreadTrack(42);
+  EXPECT_EQ(2ull, track_manager_->GetAllTracks().size());
+}
+
+TEST_F(TrackManagerTest, EmptyTracksAreNotVisible) {
+  TrackManagerTest_GetOrCreateCreatesTracks_Test();
+  EXPECT_EQ(0ull, track_manager_->GetVisibleTracks().size());
+  track_manager_->UpdateTracksForRendering();
+  EXPECT_EQ(0ull, track_manager_->GetVisibleTracks().size());
+}
+
+TEST_F(TrackManagerTest, NonEmptyTracksAreVisible) {
+  CreateAndFillSchedulerTracks();
```

Comment:
This code shouldn't compile, because this method doesn't exist. You should have a fixup unless you will combine some commits together